### PR TITLE
docs: steps for adding testing against new Kubernetes release

### PR DESCRIPTION
**Pull request by pohly**

### CODE REVIEW

File: SIDECAR_RELEASE_PROCESS.md [hunk 1]
```diff
@@ -104,3 +104,40 @@ naming convention `<hostpath-deployment-version>-on-<kubernetes-version>`.
...
+   Kubernetes release. However, jobs require less resources when they
+   don't need to build those images from the Kubernetes source code.
+   This change needs to be tried out in a PR against a component
+   first, then get submitted against csi-release-tools.
+1. Optional: propagate the updated csi-release-tools to all components
+   with the script from
+   https://github.com/kubernetes-csi/csi-release-tools/issues/7#issuecomment-707025402
+1. Once it is likely to work in all components, unset
+   `experimental_k8s_version` and submit the updated jobs.
+1. Once all sidecars for the new Kubernetes release are released,
```
Comment (msau42) [inline SIDECAR_RELEASE_PROCESS.md:134]: After sidecars are updated in host path driver, I believe we also need to cut a new version of hostpath driver, and update prow.sh/gen_jobs.sh to use it

Comment (pohly) [reply SIDECAR_RELEASE_PROCESS.md:134]: I added that.

**Comment by pohly** [comment]
https://github.com/kubernetes/test-infra/pull/20559 implements the first step (updating Prow jobs).

**Comment by pohly** [comment]
/release-note-none

We concluded the support Kubernetes 1.20 according to this checklist, so I think it is complete.

@msau42 let's merge it.

**Pull request merged**